pkg/storage/pebble.go

```diff
@@ -2034,10 +2034,30 @@ func pebbleExportToSst(
 
 		unsafeValue := iter.UnsafeValue()
 		isNewKey := !options.ExportAllRevisions || !unsafeKey.Key.Equal(curKey)
-		if paginated && options.ExportAllRevisions && isNewKey {
+		if trackKeyBoundary && options.ExportAllRevisions && isNewKey {
 			curKey = append(curKey[:0], unsafeKey.Key...)
 		}
 
+		if options.ResourceLimiter != nil {
```

Comment:
I like how much simpler this is. And if we remove the difference between hard and soft limits (other comment) it will be even simpler.

Reply:
To reiterate, export requests would prefer to get sstables split on key boundaries to avoid additional processing in tenant. So having two thresholds doesn't seem excessively complex.